cli/context/store/store.go

```diff
@@ -68,13 +84,44 @@ func New(dir string, cfg Config) Store {
 	}
 }
 
+func (s *store) WithDefaultContextInitializer(initializer DefaultContextInitializer) Store {
```

Comment:
package store is all about storing contexts (and retrieving them). It should remain pure and don't know anything about a "default" context